### ISSUE TITLE
All opened VSCode windows must be closed

### ISSUE THREAD

**Issue by bitmage0823** [microsoft/vscode#224939]
Type: <b>Feature Request</b>

When I want to see some update on command prompt right after installing some programs like python, only the solution is to close all opened VS Code windows.
Mostly people restart only the VS Code window in which the newly updated command affects the project

VS Code version: Code 1.92.0 (b1c0a14de1414fcdaa400695b4db1c0799bc3124, 2024-07-31T23:26:45.634Z)
OS version: Windows_NT x64 10.0.19042
Modes:

**Comment by bitmage0823** [comment]
Right. I need to close all the VS Code windows in order to see the new
installed program.

**Comment by albertosantini** [comment]
If I am not wrong, it is expected, because the environment variables are read when VSCode starts and "injected" to the terminal.

**Comment by bitmage0823** [comment]
I mean, only restarting the vscode doesn't make it work when multiple of
them are opened at that time.

**Comment by albertosantini** [comment]
I understand you have multiple instances of VSCode running.for different projects.

Then you install an external software and you expect, after closing the instance of VSCode related with the new software just installed, VSCode "see" the capabilities of the new software.

But this is not the result and only after closing ALL the instances of VSCode the behaviour is that expected.

Did I understand correctly?

I suspect  there is only one parent process for all instances of VSCode, sharing a few info, like enviroment variables.
So you need to close all the instances to see the changes.

**Comment by bitmage0823** [comment]
Yep exactly right.
I understand what you mean.
Ok, that was it. it was just my opinion and suggestion.
Never mind if you can't make it.
I know it requires some effort. maybe you can check if new software
installed in real-time. but not so necessary.
Thank you for your kind explanation. I'm really impressed for your kindness

**Comment by albertosantini** [comment]
Thanks. I dig a bit more.

Reading a few old issues, after searching for "environment variables", I discovered VSCode reads the changes of the environment variables, without restart, but only for system environment variables and for user ones.

I tried that (Windows):
- open VSCode
- open the terminal in VSCode
- edit the system environment variable, adding a new one, for instance
- check that variable in the terminal and you can verify it exists.
https://github.com/microsoft/vscode/issues/216563

So if the software installation adds or changes some user environment variables, and not the system ones, VSCode is blind.

**Comment by niceEli** [comment]
> Thanks. I dig a bit more.
> 
> Reading a few old issues, after searching for "environment variables", I discovered VSCode reads the changes of the environment variables, without restart, but only for system environment variables and for user ones.
> 
> I tried that (Windows):
> 
>     * open VSCode
> 
>     * open the terminal in VSCode
> 
>     * edit the system environment variable, adding a new one, for instance
> 
>     * check that variable in the terminal and you can verify it exists.
>       [VS Code in Windows doesn't seem to pull in user environment variables other than PATH #216563](https://github.com/microsoft/vscode/issues/216563)
> 
> 
> So if the software installation adds or changes some user environment variables, and not the system ones, VSCode is blind.

It seems that vscode (on windows) is only finding my system path, and not my user path, same with the other variables, only system stuff, no user stuff.
